Open Judge System/Web/OJS.Web/Areas/Contests/ViewModels/Contests/ContestViewModel.cs

```diff
@@ -116,6 +119,24 @@ public string CategoryName
         public IEnumerable<ContestCategoryListViewModel> ParentCategories { get; set; } =
             Enumerable.Empty<ContestCategoryListViewModel>();
 
+        public bool Visible
+        {
+            get
+            {
+                if (!this.IsVisible && this.VisibleFrom == null)
+                {
+                    return false;
+                }
+
+                if (this.IsVisible || (this.VisibleFrom != null && this.VisibleFrom <= DateTime.Now))
+                {
+                    return true;
+                }
+
+                return this.IsVisible;
+            }
```

Comment:
Isn't this the same check as above?

Reply:
Yes, but i think it cannot be unified to use only the field and logic from Contest.cs as the view model does not inherit the Contest class, thus why i have to copy the logic here, same as CanBeCompeted and CanBePracticed fields. 

I've thought about moving this in a method in ContestExtensions, but it has to be made Common (we currently have one in OJS.Web project).

If you have any ideas how we can change this, please advise.